plugins/out_kinesis_firehose/firehose_api.c

```diff
@@ -216,14 +217,27 @@ static int process_event(struct flb_firehose *ctx, struct flush *buf,
                          ctx->delivery_stream);
             return 2;
         }
-        /* guess space needed to write time_key */
-        len = 6 + strlen(ctx->time_key) + 6 * strlen(ctx->time_key_format);
+
+        /* format time output and return the length */
+        len = flb_aws_strftime_precision(&out_buf, ctx->time_key_format, tms);
+        if (len == -1) {
+            /*
+             * when function flb_aws_strftime_precision's return value is -1,
+             * time_key will not be added to record.
+             */
+            flb_plg_error(ctx->ins, "Failed to add time_key %s to record, %s",
+                          ctx->time_key, ctx->delivery_stream);
+            return -1;
+        }
+
```

Comment:
Please read the comment above this function, it says:

> -1 = failure, record not added

We discussed this and agreed we want to still add the record to the buffer, and just not add the timekey. This corresponds with a 0 return value.